stdlib/type.mli

```diff
@@ -35,3 +35,78 @@ type (_, _) eq = Equal: ('a, 'a) eq
 
     At runtime, this function simply returns its second argument unchanged.
 *)
+
+(** {1:identifiers Type identifiers} *)
+
+(** Type identifiers.
+
+    Type identifiers are values that can be tested for
+    {{!Id.equal}equality} to prove they represent the same type. Note
+    that unequal identifiers does not imply unequal types: the same
+    type can be represented by multiple identifiers.
+
+    See an {{!Id.example}example} of use. *)
+module Id : sig
+
+  (** {1:ids Type identifiers} *)
+
+  type 'a t
+  (** The type for type identifiers. ['a] is the represented type. *)
```

Comment:
This is consistent with the terminology introduced in the preamble. I personally wouldn't be confused by it. But if you think so, the please provide a full terminological story.

Reply:
I was thinking of having `to prove they identify the same type`, `a type can have multiple identifiers`, `the type of identifiers for ['a]`, but now I think having `represent` is good so as not to sound tautological.

Reply:
If the goal is to avoid *represent* maybe we could use *denote*. What about:

```
A type identifier is a value that denotes a type. Given two 
type identifiers they can be tested for {{!Id.equal}equality} 
to prove they denote the same type. Note that unequal identifiers
does not imply unequal types: a given type can be denoted by more
than one identifier.

type 'a t
(** The type for identifiers for type ['a]. *)
```